ca/certificates.go

```diff
@@ -579,6 +562,79 @@ func ValidateCertChain(rootPool *x509.CertPool, certs []byte, allowExpired bool)
 	return parsedCerts, nil
 }
 
+// newLocalSigner validates the signing cert and signing key to create a local signer, which accepts a crypto signer and a cert
+func newLocalSigner(keyBytes, certBytes []byte, certExpiry time.Duration, rootPool, intermediatePool *x509.CertPool) (*LocalSigner, error) {
+	switch {
+	case len(keyBytes) == 0 && len(certBytes) == 0: // not valid signer, but no error
+		return nil, nil
+	case len(keyBytes) > 0 && len(certBytes) > 0: // possibly valid signer
+	default:
+		return nil, errors.New("must provide both a signing key and a signing cert, or neither")
+	}
+
+	parsedCerts, err := helpers.ParseCertificatesPEM(certBytes)
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid signing CA cert")
+	}
+	if err := validateSignatureAlgorithm(parsedCerts[0]); err != nil {
```

Comment:
I believe `ParseCertificatesPEM` can return an empty slice if you pass in whitespace only. So a bounds check is necessary here.